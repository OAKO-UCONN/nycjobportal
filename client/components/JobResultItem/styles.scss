@import '../../assets/stylesheets/breakpointMixins';

.cardContainer {
  transition-duration: 0.2s;
  transition-property: box-shadow, transform;
  transition-timing-function: ease-in-out;

  &:hover {
    box-shadow: 4px 4px 4px rgba(0, 0, 0, 0.2);
    transform: translate(-1%, -2%);
  }

  &:active {
    box-shadow: inset 2px 2px 2px rgba(0, 0, 0, 0.4), 0 0 1px rgba(0, 0, 0, 0);
    transform: translate(0%, 0%);
  }

  @include tablet {
    margin-right: 45px;
    width: calc((100% - 90px) / 2);
    display: inline-block;
  }

  @include medium-desktop {
    margin-right: 30px;
    width: calc((100% - 60px) / 3);
  }

  &:nth-of-type(3n + 3) {
    @include medium-desktop {
      margin-right: 0px;
    }
  }

  p {
    font-size: 1.3rem;
  }

  .jobAttribute {
    color: #718096;
  }

  .agencyHeader {
    .jobAttribute {
      border-bottom: 0.1rem solid #000;
      font-size: 1.9rem;
    }
  }

  .civilServiceTitle {
    .jobAttribute {
      text-transform: capitalize;
    }
  }
}

.jobListingURL {
  text-decoration: none;
  color: inherit;
}
