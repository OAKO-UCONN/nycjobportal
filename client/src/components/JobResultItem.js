import React from 'react';
import { Box } from '@chakra-ui/core';

const JobResultItem = props => {
  const {
    agency,
    business_title,
    civil_service_title,
    division_work_unit,
    full_time_part_time_indicator,
    job_category,
    work_location,
  } = props.data;
  return (
    <div style={{ border: '1px solid red', marginBottom: '10px' }}>
      <Box>
        <p>{agency}</p>
        <p>{business_title}</p>
        <p>{civil_service_title}</p>
        <p>{division_work_unit}</p>
        <p>{full_time_part_time_indicator}</p>
        <p>{job_category}</p>
        <p>{work_location}</p>
      </Box>
    </div>
  );
};

export default JobResultItem;
