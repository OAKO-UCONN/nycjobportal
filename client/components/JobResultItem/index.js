import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { Box, Badge, Text } from '@chakra-ui/core';
import * as actions from '../../actions';
import styles from './styles.scss';

const formatFullTimePartTimeIndicator = fullTimePartTimeIndicator => {
  if (fullTimePartTimeIndicator === 'F') return 'Full Time';
  if (fullTimePartTimeIndicator === 'P') return 'Part Time';
  return null;
};

const JobResultItem = props => {
  const { data } = props;

  const {
    agency,
    job_id: jobID,
    posting_type: postingType,
    business_title: businessTitle,
    civil_service_title: civilServiceTitle,
    division_work_unit: divisionWorkUnit,
    full_time_part_time_indicator: fullTimePartTimeIndicator,
    job_category: jobCategory,
    work_location: workLocation,
  } = data;

  const handleOnClick = props => {
    const { fetchSingleJob, data } = props;
    const { job_id: jobID, posting_type: jobPostingType } = data;
    fetchSingleJob(jobID, jobPostingType);
  };

  return (
    <Box
      className={styles.cardContainer}
      rounded="20px"
      boxShadow="sm"
      border="0.1rem solid"
      borderColor="gray.500"
      mb="3.25rem"
    >
      <Link
        className={styles.jobListingURL}
        onClick={() => handleOnClick(props)} //eslint-disable-line
        to={{
          pathname: `/job-listing/${jobID}?id=${jobID}&postingType=${postingType}`,
        }}
      >
        <Box p={5}>
          <Badge p={2} variant="outline" mr="1rem">
            {postingType}
          </Badge>
          {fullTimePartTimeIndicator && (
            <Badge p={2} variant="outline">
              {formatFullTimePartTimeIndicator(fullTimePartTimeIndicator)}
            </Badge>
          )}
          <Text className={styles.agencyHeader}>
            <span className={styles.jobAttribute}>{agency}</span>
          </Text>
          <Text>
            Business Title:{' '}
            <span className={styles.jobAttribute}>{businessTitle}</span>
          </Text>
          <Text className={styles.civilServiceTitle}>
            Civil Service Title:{' '}
            <span className={styles.jobAttribute}>
              {civilServiceTitle.toLowerCase()}
            </span>
          </Text>
          <Text>
            Work Unit:{' '}
            <span className={styles.jobAttribute}>{divisionWorkUnit}</span>
          </Text>
          <Text>
            Job Category:{' '}
            <span className={styles.jobAttribute}>{jobCategory}</span>
          </Text>
          <Text>
            Location:{' '}
            <span className={styles.jobAttribute}>{workLocation}</span>
          </Text>
        </Box>
      </Link>
    </Box>
  );
};

export default connect(null, actions)(JobResultItem);
