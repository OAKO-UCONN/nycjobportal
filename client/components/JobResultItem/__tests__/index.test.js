import React from 'react';
import mockJobItem from '../__mocks__/mockJobItem';
import JobResultItem from '../index';

describe('JobResultItem', () => {
  describe('optimal case', () => {
    const { asFragment } = reduxConnectedRender(
      <JobResultItem data={mockJobItem} />
    );

    it('should match snapshot', () => {
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
