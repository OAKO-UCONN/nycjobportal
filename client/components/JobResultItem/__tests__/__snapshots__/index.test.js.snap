// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobResultItem optimal case should match snapshot 1`] = `
<DocumentFragment>
  .emotion-9 {
  border-radius: 20px;
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.1),0 1px 2px 0 rgba(0,0,0,0.06);
  border: 0.1rem solid;
  border-color: #718096;
  margin-bottom: 3.25rem;
}

.emotion-8 {
  padding: 1.25rem;
}

.emotion-0 {
  display: inline-block;
  padding-left: 0.25rem;
  padding-right: 0.25rem;
  text-transform: uppercase;
  font-size: 0.75rem;
  border-radius: 0.125rem;
  font-weight: 700;
  white-space: nowrap;
  vertical-align: middle;
  box-shadow: inset 0 0 0px 1px #718096;
  color: #718096;
  padding: 0.5rem;
  margin-right: 1rem;
}

.emotion-1 {
  display: inline-block;
  padding-left: 0.25rem;
  padding-right: 0.25rem;
  text-transform: uppercase;
  font-size: 0.75rem;
  border-radius: 0.125rem;
  font-weight: 700;
  white-space: nowrap;
  vertical-align: middle;
  box-shadow: inset 0 0 0px 1px #718096;
  color: #718096;
  padding: 0.5rem;
}

.emotion-2 {
  font-family: Roboto,-apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
}

<div
    class="cardContainer emotion-9"
  >
    <a
      class="jobListingURL"
      href="/job-listing/438977-Internal"
    >
      <div
        class="emotion-8"
      >
        <div
          class="emotion-0"
        >
          Internal
        </div>
        <div
          class="emotion-1"
        >
          Full Time
        </div>
        <p
          class="agencyHeader emotion-2"
        >
          <span
            class="jobAttribute"
          >
            DEPT OF PARKS & RECREATION
          </span>
        </p>
        <p
          class="emotion-2"
        >
          Business Title: 
          <span
            class="jobAttribute"
          >
            PEP Step Up Captain
          </span>
        </p>
        <p
          class="civilServiceTitle emotion-2"
        >
          Civil Service Title: 
          <span
            class="jobAttribute"
          >
            community coordinator
          </span>
        </p>
        <p
          class="emotion-2"
        >
          Work Unit: 
          <span
            class="jobAttribute"
          >
            Arsenal North Peps
          </span>
        </p>
        <p
          class="emotion-2"
        >
          Job Category: 
          <span
            class="jobAttribute"
          >
            Public Safety, Inspections, & Enforcement
          </span>
        </p>
        <p
          class="emotion-2"
        >
          Location: 
          <span
            class="jobAttribute"
          >
            105 St & 5 Ave
          </span>
        </p>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
